File: flutter17_mobile/lib/models/gaming_console.dart
import 'package:json_annotation/json_annotation.dart';
import 'product.dart';

part 'gaming_console.g.dart';

@JsonSerializable()
class GamingConsole {
  int? id;
  int? productId;

  // General Information        
  String? processor;
  String? graphicsProcessor;

  // Performance
  int? ram; // RAM size in GB
  String? storageType;
  int? storageCapacity; // Storage size in GB or TB
  String? maxResolution;
  int? maxFPS; // Maximum Frames Per Second

  // Connectivity
  int? usbPorts;
  bool? hasWiFi;
  bool? hasBluetooth;
  bool? hasEthernetPort;
  bool? supportsExternalStorage;

  // Features
  bool? supportsVR;
  bool? hasPhysicalMediaDrive;
  bool? isPortable;

  // Additional Features
  String? controllerType;
  bool? supportsBackwardCompatibility;
  String? onlineService;

  // Dimensions and Weight
  double? weight;
  String? dimensions;
  
  GamingConsole({
    this.id,
    this.productId,
    this.processor,
    this.graphicsProcessor,
    this.ram,
    this.storageType,
    this.storageCapacity,
    this.maxResolution,
    this.maxFPS,
    this.usbPorts,
    this.hasWiFi,
    this.hasBluetooth,
    this.hasEthernetPort,
    this.supportsExternalStorage,
    this.supportsVR,
    this.hasPhysicalMediaDrive,
    this.isPortable,
    this.controllerType,
    this.supportsBackwardCompatibility,
    this.onlineService,
    this.weight,
    this.dimensions,
  });

  factory GamingConsole.fromJson(Map<String, dynamic> json) =>
      _$GamingConsoleFromJson(json);

  Map<String, dynamic> toJson() => _$GamingConsoleToJson(this);
}

File: angular17_web/src/app/components/see-more-details/spec-builders/gaming-console.ts
import { boolRow, row, SpecSection } from './shared';

export function gamingConsoleSections(p: any): SpecSection[] {
  return [
    {
      title: 'General Information',
      rows: [
        row('Processor', p.processor),
        row('Graphics Processor', p.graphicsProcessor),
      ],
    },
    {
      title: 'Performance',
      rows: [
        row('RAM', p.ram, ' GB'),
        row('Storage Type', p.storageType),
        row('Storage Capacity', p.storageCapacity, ' GB'),
        row('Max Resolution', p.maxResolution),
        row('Max FPS', p.maxFPS),
      ],
    },
    {
      title: 'Connectivity',
      rows: [
        row('USB Ports', p.usbPorts),
        boolRow('WiFi', p.hasWiFi),
        boolRow('Bluetooth', p.hasBluetooth),
        boolRow('Ethernet Port', p.hasEthernetPort),
        boolRow('External Storage', p.supportsExternalStorage),
      ],
    },
    {
      title: 'Features',
      rows: [
        boolRow('VR', p.supportsVR),
        boolRow('Physical Media Drive', p.hasPhysicalMediaDrive),
        boolRow('Portable', p.isPortable),
      ],
    },
    {
      title: 'Additional Features',
      rows: [
        row('Controller Type', p.controllerType),
        boolRow('Backward Compatibility', p.supportsBackwardCompatibility),
        row('Online Service', p.onlineService),
      ],
    },
    {
      title: 'Physical Characteristics',
      rows: [
        row('Weight', p.weight, ' kg'),
        row('Dimensions', p.dimensions),
      ],
    },
  ];
}
